src/Collection.php

```diff
@@ -291,13 +291,17 @@ public function duplicate(): CollectionInterface
     /**
      * Create a new instance with no items.
      *
-     * @psalm-template NewTKey
-     * @psalm-template NewTKey of array-key
-     * @psalm-template NewT
+     * @template NewTKey of array-key
+     * @template NewT
+     *
+     * @return self<NewTKey, NewT>
      */
     public static function empty(): CollectionInterface
     {
-        return self::fromIterable([]);
+        /** @var array<NewTKey, NewT> $emptyArray */
+        $emptyArray = [];
+
+        return self::fromIterable($emptyArray);
```

Comment:
this is the only way to get around having template parameters for the empty array, otherwise PHPStan complains because it cannot figure out the template types:
```
  301    Method loophp\collection\Collection::empty() should
         return loophp\collection\Collection<NewTKey of
         (int|string), NewT> but returns
         loophp\collection\Collection<*NEVER*, *NEVER*>.

Reply:
Should we report that as a bug ?

Reply:
yeah leave it with me. Looks like a valid issue though maybe I thought it makes sense because it has no way to know where the template parameters are coming from

Reply:
done! https://github.com/phpstan/phpstan/issues/5065